src/js/intlTelInput.js

```diff
@@ -1063,7 +1058,8 @@ Plugin.prototype = {
   _beforeSetNumber: function(number) {
     if (this.options.separateDialCode) {
       var dialCode = this._getDialCode(number);
-      if (dialCode) {
+      // _getDialCode returns empty string if there is no dialCode
+      if (dialCode && dialCode !== "") {
```

Comment:
FYI this second check is redundant - `if (dialCode) {` will already be falsy if dialCode is an empty string - most likely I use this technique in a few places throughout the code.